src/features/testingProvider.ts

```diff
@@ -0,0 +1,983 @@
+                d3,
+                d4,
+                d5,
+                d6,
+                d7,
+                new Disposable(s1),
+                new Disposable(s2)
+            )
+        );
+    }
+
+    public resolveTestCase(assembly: string, id: string): TestCase | undefined {
+        return this._testAssemblies.get(assembly)?.discoveredTests.get(id);
+    }
+
+    public getAllTestCases(): TestCase[] {
+        const testCase = [];
+        for (const assembly of this._testAssemblies.values()) {
+            testCase.push(...assembly.discoveredTests.values());
+        }
+        return testCase;
+    }
+
+    public reportFileChange(fileName: string) {
+        this._fileChangeDebouncer.next(fileName);
+    }
+
+    private async _reportFileChanges(fileNames: string[]): Promise<void> {
+        for (const fileName of new Set(fileNames)) {
+            let projectInfo: ProjectInformationResponse;
+            try {
+                projectInfo = await serverUtils.requestProjectInformation(
+                    this._server,
+                    { FileName: fileName }
+                );
+            } catch (error) {
+                return undefined;
+            }
+            this.reportProject(projectInfo.MsBuildProject);
+        }
+    }
+
+    public async removeProject(project: MSBuildProject): Promise<void> {
+        this._testAssemblies.delete(project.AssemblyName);
+        this.controller.items.delete(project.AssemblyName);
+    }
+
+    public reportProject(project: MSBuildProject) {
+        this._projectChangedDebouncer.next(project);
+    }
+
+    private async _discoverTests(
+        fileName: string,
+        testFramework: string,
+        targetFramework: string
+    ) {
+        let testInfo: V2.TestInfo[] | undefined = undefined;
+        try {
+            testInfo = await this.testManager.discoverTests(
+                fileName,
+                testFramework,
+                false,
+                targetFramework
+            );
+        } catch {}
+        if (testInfo == undefined) {
+            testInfo = await this.testManager.discoverTests(
+                fileName,
+                testFramework,
+                true,
+                targetFramework
+            );
+        }
+
+        return testInfo;
+    }
+
+    private async _reportProject(project: MSBuildProject): Promise<void> {
+        if ((project.SourceFiles?.length ?? 0) == 0) {
+            return;
+        }
+
+        this._eventStream.post(
+            new DotNetTestDiscoveryStart(project.AssemblyName)
+        );
+
+        let testInfo: (V2.TestInfo & { targetFramework: string })[] = [];
+
+        const start = Date.now();
+        await Promise.all(
+            project.TargetFrameworks.map(async (targetFramework) => {
+                testInfo.push(
+                    ...(
+                        await this._discoverTests(
+                            project.SourceFiles[0],
+                            "xunit",
```

Comment:
This should be dynamic, but I haven't found a good way to find out what the used testframework is.

Reply:
Crazy as it may sound. It looks like the framework is ignored on the O# side. See https://github.com/OmniSharp/omnisharp-roslyn/blob/ff21ac475cdf8afb15579f2b5fb346945db2a7e9/src/OmniSharp.DotNetTest/VSTestManager.cs#L130-L144

Also, starting and debugging tests do validate that the test framework name is one known to O#. It doesn't seem to play into actually starting or debugging the tests.

Reply:
a sweet! Resolved :) 
![image](https://user-images.githubusercontent.com/14233220/153524651-2cada842-86e3-4432-ae81-a16f75644cb3.png)